tests/unit/fixtures/gh-738.js

```diff
@@ -0,0 +1,31 @@
+/* jshint browser:true */
+/* jshint node:true */
+
+// object with "eval" key
+var obj = {
+	eval: function (str) {
+		return str;
+	},
+	wrapper: function (str) {
+		// method calling "eval" key from context
+		// permitted use
+		return this.eval(str);
+	}
+};
+
+// object-key use, permitted
+obj["eval"]('console.log(\'hello world\');');
```

Comment:
Using double quotes consistently here and below will be more in-line with the project's code style and will allow you to avoid escaping quote marks.